collectors/framework/framework.go

```diff
@@ -47,6 +43,32 @@ type AvroDatum struct {
 	ApproxBytes int64
 }
 
+// Collector defines the structure of the framework collector and facilitates
+// mocking the various configuration options during testing.
+type Collector struct {
+	listenEndpointFlag         string
+	recordInputLogFlag         bool
+	inputLimitAmountKBytesFlag int
+	inputLimitPeriodFlag       int
+}
+
+// countingReader is an io.Reader that provides counts of the number of bytes
+// read, and which supports optional hexdumps of the data that it's reading.
+type countingReader struct {
+	readerImpl io.Reader
+	inputBytes int64
+}
+
+// New returns a new instance of the framework collector.
+func New() Collector {
+	return Collector{
+		listenEndpointFlag:         "127.0.0.1:8124",
+		recordInputLogFlag:         false,
+		inputLimitAmountKBytesFlag: 20480,
+		inputLimitPeriodFlag:       60,
+	}
+}
```

Comment:
This allows us to use `Collector{}` directly and mock these values during testing. But otherwise, calling `framework.New()` returns all the defaults we were accustomed to in `main`.